tools/src/wyvern/target/corewyvernIL/type/RefinementType.java

```diff
@@ -106,14 +145,30 @@ public ValueType adapt(View v) {
         return new RefinementType(newBase, newDTs, this, selfSite);
     }
 
+    private static GenericArgument adaptArgument(View v, GenericArgument ga) {
```

Comment:
This method seems to belong to the `GenericArgument` class instead. Please move it there.